certbot/ocsp.py

```diff
@@ -0,0 +1,136 @@
+"""Tools for checking certificate revocation."""
+import logging
+
+from subprocess import Popen, PIPE
+
+from certbot import errors
+from certbot import util
+
+logger = logging.getLogger(__name__)
+
+
+REV_LABEL = "REVOKED"
+
+INSTALL_LABEL = "(Installed)"
+
+class RevocationChecker(object):
+    "This class figures out OCSP checking on this system, and performs it."
+
+    def __init__(self, config):
+        self.broken = False
+        self.config = config
+
+        if not util.exe_exists("openssl"):
+            logging.info("openssl not installed, can't check revocation")
+            self.broken = True
+            return
+
+       # New versions of openssl want -header var=val, old ones want -header var val
+        test_host_format = Popen(["openssl", "ocsp", "-header", "var", "val"],
+                                 stdout=PIPE, stderr=PIPE, universal_newlines=True)
+        _out, err = test_host_format.communicate()
+        if "Missing =" in err:
+            self.host_args = lambda host: ["Host=" + host]
+        else:
+            self.host_args = lambda host: ["Host", host]
+
+
+    def ocsp_status(self, cert_path, chain_path, status_in):
+        """Helper function: updates a cert status string with revocation information
+
+        :param str cert_path: path to a cert to check
+        :param str chain_path: issuing intermediate for the cert
+        :param str status_in: a string that is either empty, if the cert is otherwise
+                              believed to be valid, or 'INVALID: $REASON'.
+
+        :returns: a new status including revocation, if the cert is revoked."""
+
+        if self.config.check_ocsp.lower() == "never":
+            return status_in
+        elif self.config.check_ocsp.lower() == "lazy" and status_in:
```

Comment:
this check can go in `cert_manager` if you really want it to exist, then I think we don't need this function at all.